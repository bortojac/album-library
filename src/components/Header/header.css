.header {
  background-color: #fff;
  display: flex;
  border-bottom: solid .2rem #000;
}

.header__h1 {
  margin-right: auto;
  margin-left: .5rem;
  margin-top: 1rem;
  margin-bottom: 1rem;
  font-size: 1.5rem;
}

.header__p {

}

