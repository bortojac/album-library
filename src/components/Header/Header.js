import React, { Component } from 'react';
import './header.css';

class Header extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.openSearch();
  }

  render() {
    return (
      <div className="header">
        <h1 className="header__h1">Album Library</h1>
        <p className="header__icon" onClick={this.handleClick}>Click Me</p>
      </div>
    );
  }
}

export default Header;