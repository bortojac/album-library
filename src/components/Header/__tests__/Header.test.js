import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from '../Header';

// check render
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot
it('renders correctly', () => {
  const tree = renderer
    .create(<Header />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
