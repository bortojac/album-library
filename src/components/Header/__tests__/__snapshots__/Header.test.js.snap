// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header"
>
  <h1
    className="header__h1"
  >
    Album Library
  </h1>
  <p
    className="header__icon"
  >
    Click Me
  </p>
</div>
`;
